scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -51,9 +82,106 @@ object BeamOp extends Serializable {
+          override def expand(input: PCollection[(K, V)]): PCollection[(K, U)] = {
+            val groupedSortedValues = input
+              .apply(TupleToKV[K, V](OrderedSerializationCoder(ordK, kryoCoder), OrderedSerializationCoder(ordV, kryoCoder)))
+              .apply(GroupByKey.create[K, V]()).setCoder(KvCoder.of(OrderedSerializationCoder(ordK, kryoCoder), IterableCoder.of(OrderedSerializationCoder(ordV, kryoCoder))))
+              .apply(SortGroupedValues[K, V])
+
+            planMapGroup[K, V, U](groupedSortedValues, reduceFn)
+              .apply(ParDo.of(FlatMapFn[KV[K, java.lang.Iterable[U]], KV[K, U]]{ elem =>
+                elem.getValue.asScala.map(KV.of(elem.getKey, _))
+              })).setCoder(KvCoder.of(OrderedSerializationCoder(ordK, kryoCoder), kryoCoder))
+              .apply(KVToTuple[K, U])
+          }
+        },
+        kryoCoder)
+    }
+
+    def sorted(implicit ordK: Ordering[K], ordV: Ordering[V], kryoCoder: KryoCoder): BeamOp[(K, V)] = {
+      TransformBeamOp[(K, V), (K, V)](
+        op,
+        new PTransform[PCollection[(K, V)], PCollection[(K, V)]]() {
+          override def expand(input: PCollection[(K, V)]): PCollection[(K, V)] = {
+            input
+              .apply(TupleToKV[K, V](OrderedSerializationCoder(ordK, kryoCoder), OrderedSerializationCoder(ordV, kryoCoder)))
+              .apply(GroupByKey.create[K, V]()).setCoder(KvCoder.of(OrderedSerializationCoder(ordK, kryoCoder), IterableCoder.of(OrderedSerializationCoder(ordV, kryoCoder))))
+              .apply(SortGroupedValues[K, V])
+              .apply(ParDo.of(FlatMapFn[KV[K, java.lang.Iterable[V]], KV[K, V]] { elem =>
+                elem.getValue.asScala.map(x => KV.of(elem.getKey, x))
+              })).setCoder(KvCoder.of(OrderedSerializationCoder(ordK, kryoCoder), OrderedSerializationCoder(ordV, kryoCoder)))
+              .apply(KVToTuple[K, V])
+          }
+        },
+        kryoCoder)
+    }
+
+    def mapSideAggregator(size: Int, semigroup: Semigroup[V])(implicit kryoCoder: KryoCoder): BeamOp[(K, V)] = {
+      TransformBeamOp[(K, V), (K, V)](
+        op,
+        new PTransform[PCollection[(K, V)], PCollection[(K, V)]]() {
+          override def expand(input: PCollection[(K, V)]): PCollection[(K, V)] =
+            input.apply(ParDo.of(MapSideAggregator[K, V](size, semigroup))).setCoder(kryoCoder)
+        },
+        kryoCoder)
+    }
+  }
+
+  case class SortGroupedValues[K, V](implicit ordK: Ordering[K], ordV: Ordering[V], kryoCoder: KryoCoder) extends PTransform[PCollection[KV[K, java.lang.Iterable[V]]], PCollection[KV[K, java.lang.Iterable[V]]]] {
+    override def expand(input: PCollection[KV[K, lang.Iterable[V]]]): PCollection[KV[K, lang.Iterable[V]]] = {
+      input
+        .apply(ParDo.of(MapFn[KV[K, java.lang.Iterable[V]], KV[K, java.lang.Iterable[V]]]{ elem =>
+          KV.of(elem.getKey, elem.getValue.asScala.toSeq.sorted.toIterable.asJava)
```

Comment:
Making it `toArray` would be more performant I think, using raw java here would be even better. But I'm fine to keep any way, since I hope we'll rewrite this eventually.

Please put `TODO` here documenting this needs to be changed to external sorter like the one in beam